src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRConnectionInputBinding/SignalRConnectionInputBinding.cs

```diff
@@ -34,38 +34,33 @@ internal class SignalRConnectionInputBinding : BindingBase<SignalRConnectionInfo
         {
             var azureSignalRClient = Utils.GetAzureSignalRClient(attrResolved.ConnectionStringSetting, attrResolved.HubName);
 
-            if (!bindingData.ContainsKey(HttpRequestName) || securityTokenValidator == null)
-            {
-                var info = await azureSignalRClient.GetClientConnectionInfoAsync(attrResolved.UserId, attrResolved.IdToken,
-                    attrResolved.ClaimTypeList);
-                return new SignalRConnectionInfoValueProvider(info, userType, "");
-            }
-
-            var request = bindingData[HttpRequestName] as HttpRequest;
-
-            var tokenResult = securityTokenValidator.ValidateToken(request);
+            bindingData.TryGetValue(HttpRequestName, out var requestObj);
+            var request = requestObj as HttpRequest;
+            var httpContext = request?.HttpContext;
 
-            if (tokenResult.Status != SecurityTokenStatus.Valid)
+            if (bindingData.ContainsKey(HttpRequestName) && securityTokenValidator != null)
             {
-                return new SignalRConnectionInfoValueProvider(null, userType, "");
-            }
+                var tokenResult = securityTokenValidator.ValidateToken(request);
 
-            if (signalRConnectionInfoConfigurer == null)
-            {
-                var info = await azureSignalRClient.GetClientConnectionInfoAsync(attrResolved.UserId, attrResolved.IdToken,
-                    attrResolved.ClaimTypeList);
-                return new SignalRConnectionInfoValueProvider(info, userType, "");
+                if (tokenResult.Status != SecurityTokenStatus.Valid)
+                {
+                    return new SignalRConnectionInfoValueProvider(null, userType, "");
+                }
+
+                var signalRConnectionDetail = new SignalRConnectionDetail
+                {
+                    UserId = attrResolved.UserId,
+                    Claims = azureSignalRClient.GetCustomClaims(attrResolved.IdToken, attrResolved.ClaimTypeList),
```

Comment:
no need to `GetCustomClaims` when `signalRConnectionInfoConfigurer` is null

Reply:
Just simplify the code, they can use the same logic branch.